python/pyiceberg/typedef.py

```diff
@@ -0,0 +1,20 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
These were moved to this new file to avoid this circular import error:
```py
ImportError while loading conftest '/Users/sam/iceberg/python/tests/conftest.py'.
tests/conftest.py:55: in <module>
    from tests.catalog.test_base import InMemoryCatalog
tests/catalog/test_base.py:28: in <module>
    from pyiceberg.catalog import (
pyiceberg/catalog/__init__.py:25: in <module>
    from pyiceberg.table.base import Table
pyiceberg/table/base.py:23: in <module>
    from pyiceberg.catalog import Identifier
E   ImportError: cannot import name 'Identifier' from partially initialized module 'pyiceberg.catalog' (most likely due to a circular import) (/Users/sam/iceberg/python/pyiceberg/catalog/__init__.py)
```